optaplanner-core/src/test/java/org/optaplanner/core/api/score/stream/ConstraintCollectorsTest.java

```diff
@@ -1381,6 +1598,224 @@ public void maxNotComparable() {
         assertResult(collector, container, null);
     }
 
+    @Test
+    public void maxComparableMapped() {
+        UniConstraintCollector<Class, ?, String> collector = max(Class::getCanonicalName);
+        Object container = collector.supplier().get();
+        // add first value, which becomes the max
+        Class firstValue = ConstraintCollectorsTest.class;
```

Comment:
A Class as a firstValue?
Wouldn't it make more sense to either use:
Object o1 = new Object();
or use mockito's "mock()").

Or does it need to be a class instance? In that case, let's call it firstClass?

Reply:
The reason I use class is that it doesn't implement `Comparable`, yet it has an easily accessible`String` field that does, so it's easy to write a `Comparator` for it.